synapse/handlers/oidc_handler.py

```diff
@@ -1213,16 +1210,23 @@ async def map_user_attributes(
             # a usable mxid.
             localpart += str(failures) if failures else ""
 
-        display_name = None  # type: Optional[str]
-        if self._config.display_name_template is not None:
-            display_name = self._config.display_name_template.render(
-                user=userinfo
-            ).strip()
+        def render_template_field(template: Optional[Template]) -> Optional[str]:
+            if template is None:
+                return None
+            return template.render(user=userinfo).strip()
+
+        display_name = render_template_field(self._config.display_name_template)
+        if display_name == "":
+            display_name = None
 
-            if display_name == "":
-                display_name = None
+        emails = []  # type: List[str]
+        email = render_template_field(self._config.email_template)
+        if email:
+            emails.append(email)
 
-        return UserAttributeDict(localpart=localpart, display_name=display_name)
+        return UserAttributeDict(
+            localpart=localpart, display_name=display_name, emails=emails
+        )
```

Comment:
FWIW I had wonder if we should just return `synapse.handlers.sso.UserAttributes` here instead of having a class that is almost identical. 🤷 I think I had mypy issues with it though and couldn't easily fix it.

No change necessary, just idly thinking. :)

Reply:
I had the same thought. Then I realised this was an extension module, so changing the expected return type could break existing implementations, and suddenly I lost enthusiasm...

Reply:
I'm pretty sure that was what I did as well. 😟